src/store/actions/roomStateAction.ts

```
@@ -1,12 +1,40 @@
-import { createAction } from '@reduxjs/toolkit';
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { IRoomState } from '@/utils/interfaces';
+import { history } from '@/utils/history';
+import { delay, getRoomIdByUrl } from '@/helpers';
 
-interface IRoomState {
+interface IRoomSettings {
   roomTitle: string;
-  voteSystem: string;
-  dealerRights: string;
-  roomId: string;
+  voteSystem?: string;
+  dealerRights?: string
 }
 
 export const roomStateActions = {
   setRoomState: createAction<IRoomState>('[ROOM_STATE]:setRoomState'),
+  getRoomByUrl: createAsyncThunk('[ROOM_STATE]:getRoomByUrl',
+    async (_: void, { rejectWithValue }) => {
+      try {
+        // const idFromUrl = getRoomIdByUrl();
+        const roomId: string = await delay('11', 100);
+        if (!roomId) {
+          history.push('/newGame');
+          return rejectWithValue('Error witn room');
```

could you throw an error and call `history.push` only in catch?

---

refactored